ebean-querybean/src/main/java/io/ebean/typequery/PBaseString.java

```diff
@@ -5,7 +5,7 @@
  *
  * @param <R> the root query bean type
  */
-public abstract class PBaseString<R,T> extends PBaseCompareable<R, String> {
+public abstract class PBaseString<R, T> extends PBaseComparable<R, T> {
```

Comment:
What was the intention of this change?
I see, that you compare Cidr and other with PBaseString.

Reply:
The intention is for types like Cdir to nicely support operators (eq, ne, like etc) with both their natural T type (Cdir) and String.

Given a cdir type we want to:

```java
.cdir.eq(myCdirValue)
```
and also just use strings (especially for like, startsWith, contains etc)  ...
```java
.cdir.eq("some-string-value")
```

... but let me check this.

Reply:
Yes, I'm happy with this.  Effectively we overload all the types that extend PBaseString so that they can take as parameters String or their type.

For example with Inet we have both `eq(Inet value)`  and `eq(String value)`

This deprecates the odd existing methods `equalToType(T value)` and `notEqualToType(T value)`